app/src/main/java/org/dslul/openboard/inputmethod/latin/settings/UserDictionaryAddWordFragment.java

```diff
@@ -115,12 +116,20 @@ private BitmapDrawable getBitmapFromVectorDrawable(int drawable, float scale) {
     @Override
     public boolean onOptionsItemSelected(MenuItem item) {
         if (item.getItemId() == OPTIONS_MENU_ADD) {
-            mContents.apply(requireActivity());
-            requireActivity().onBackPressed();
+            if (mContents.isExistingWord(requireContext())) {
+                new AlertDialog.Builder(requireContext())
+                        .setMessage(R.string.user_dict_word_already_present)
+                        .setPositiveButton(android.R.string.ok, (dialog, i) -> mContents.editWord(requireContext()))
+                        .setNegativeButton(android.R.string.cancel, (dialog, i) -> requireActivity().onBackPressed())
```

Comment:
`requireActivity().onBackPressed()` means that the add/edit screen will be closed, right? This might be unexpected from just not wanting to overwrite.
If you agree, you could also use the `confirmDialog` for this.

Reply:
> requireActivity().onBackPressed() means that the add/edit screen will be closed, right?

Exactly and the user go back to the previous.

> If you agree, you could also use the `confirmDialog` for this.

I'm not against using confimDialog, but Android Studio gives me this error:

- Required type:Function0<kotlin.Unit>
- Make `editWord` return kotlin.jvm.functions.Function0<kotlin.Unit>

Do I need to validate?

Reply:
Oh, I didn't expect that... not even returning `Unit.INSTANCE` works. Then better leave it a normal dialog.

> Exactly and the user go back to the previous.

I think this should not happen. Typical behavior of canceling is just that nothing happens, which would allow the user to check again and then go back manually, or maybe fix some typo and try again.

Reply:
> I think this should not happen. Typical behavior of canceling is just that nothing happens, which would allow the user to check again and then go back manually, or maybe fix some typo and try again.

No problem; I'll make the necessary changes.

Reply:
With the commit 21ef30f it should be better.